pkg/env/var.go

```diff
@@ -0,0 +1,178 @@
+		allVars[v.Name] = v // last one with a description wins if there are multiple uses of the same variable in a single process
+	}
+
+	mutex.Unlock()
+
+	if result, ok := os.LookupEnv(v.Name); ok {
+		return result, true
+	}
+
+	return v.DefaultValue, false
+}
+
+// GetComplexVarAsBool returns the value of a boolean environment variable, or a default value if the environment variable isn't set.
+//
+// In addition to the value, this function returns a boolean value indicating whether the variable existed or not.
+func GetComplexVarAsBool(v Var) (bool, bool) {
+	result, ok := GetComplexVar(v)
+
+	b, err := strconv.ParseBool(result)
+	if err != nil {
+		log.Warnf("Invalid environment variable value `%s`, expecting true/false, defaulting to %v", result, v.DefaultValue)
+		b, _ = strconv.ParseBool(v.DefaultValue)
+	}
+
+	return b, ok
+}
+
+// GetComplexVarAsInt returns the value of an integer environment variable, or a default value if the environment variable isn't set.
+//
+// In addition to the value, this function returns a boolean value indicating whether the variable existed or not.
+func GetComplexVarAsInt(v Var) (int, bool) {
+	result, ok := GetComplexVar(v)
+
+	i, err := strconv.Atoi(result)
+	if err != nil {
+		log.Warnf("Invalid environment variable value `%s`, expecting an integer, defaulting to %v", result, v.DefaultValue)
+		i, _ = strconv.Atoi(v.DefaultValue)
+	}
+
+	return i, ok
+}
+
+// GetComplexVarAsFloat returns the value of a floating-point environment variable, or a default value if the environment variable isn't set.
+//
+// In addition to the value, this function returns a boolean value indicating whether the variable existed or not.
+func GetComplexVarAsFloat(v Var) (float64, bool) {
+	result, ok := GetComplexVar(v)
+
+	f, err := strconv.ParseFloat(result, 64)
+	if err != nil {
+		log.Warnf("Invalid environment variable value `%s`, expecting a floating-point value, defaulting to %v", result, v.DefaultValue)
+		f, _ = strconv.ParseFloat(v.DefaultValue, 64)
+	}
+
+	return f, ok
+}
+
+// GetComplexVarAsDuration returns the value of a duration environment variable, or a default value if the environment variable isn't set.
+//
+// In addition to the value, this function returns a boolean value indicating whether the variable existed or not.
+func GetComplexVarAsDuration(v Var) (time.Duration, bool) {
+	result, ok := GetComplexVar(v)
+
+	d, err := time.ParseDuration(result)
+	if err != nil {
+		log.Warnf("Invalid environment variable value `%s`, expecting a duration, defaulting to %v", result, v.DefaultValue)
+		d, _ = time.ParseDuration(v.DefaultValue)
+	}
+
+	return d, ok
+}
+
+// GetVar returns the value of an environment variable, or a default value if the environment variable isn't set.
+//
+// In addition to the value, this function returns a boolean value indicating whether the variable existed or not.
+func GetVar(name string, defaultValue string, description string) (string, bool) {
```

Comment:
This name stutters `var.GetVar` ... should it just be `Get`, or maybe `GetString`?

Same thing with all the `GetVarXXX` methods ... they can just be `GetXXX`.

Reply:
The package name is env. So that makes env.GetVar which sounds right.